jetbrains-core/tst/software/aws/toolkits/jetbrains/core/region/MockRegionProvider.kt

```diff
@@ -102,14 +84,31 @@ class MockRegionProviderRule : ExternalResource() {
         )
     )
 
-    fun addService(serviceName: String, service: Service) = regionManager.addService(serviceName, service)
+    fun createAwsRegion(id: String = uniqueRegionId(), partitionId: String = aString()): AwsRegion =
+        anAwsRegion(id = id, partitionId = partitionId).also { regionManager.addRegion(it) }
 
-    override fun after() {
-        reset()
+    private fun uniqueRegionId(): String {
+        repeat(10) {
+            val generatedId = aRegionId()
+            if (regionManager[generatedId] == null) {
+                return generatedId
+            }
+        }
+        throw IllegalStateException("Failed to generate a unique region ID")
     }
 
+    fun defaultPartition(): AwsPartition = regionManager.defaultPartition()
+
+    fun defaultRegion(): AwsRegion = regionManager.defaultRegion()
+
+    fun addService(serviceName: String, service: Service) = regionManager.addService(serviceName, service)
+
     fun reset() {
         @Suppress("DEPRECATION")
         regionManager.reset()
     }
 }
+
+// dynamically get the default region from whatever is currently registered
+@TestOnly
```

Comment:
Test only isnt needed if its in a test file

Reply:
also can be shorted to `service<ToolkitRegionProvider>()`

Reply:
This was moved around a few times so it was lingering, removed